pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -337,9 +372,52 @@ def build_model(
                 dims="date",
             )
 
-            intercept = self.intercept_dist(
-                name="intercept", **self.model_config["intercept"]["kwargs"]
-            )
+            if self.time_varying_intercept or self.time_varying_media_effect:
+                time_index = pm.MutableData(
+                    "time_index",
+                    self._time_index,
+                    dims="date",
+                )
+
+            if self.time_varying_intercept:
+                tv_multiplier_intercept = time_varying_prior(
+                    name="tv_multiplier_intercept",
+                    X=time_index,
+                    X_mid=self._time_index_mid,
+                    positive=True,
+                    m=200,
+                    L=[self._time_index_mid + 365 / self._time_resolution],
+                    ls_mu=365 / self._time_resolution * 2,
+                    ls_sigma=10,
+                    eta_lam=1,
+                    dims="date",
+                )
+                intercept_base = self.intercept_dist(
+                    name="intercept_base", **self.model_config["intercept"]["kwargs"]
+                )
+                intercept = pm.Deterministic(
+                    name="intercept",
+                    var=intercept_base * tv_multiplier_intercept,
+                    dims="date",
+                )
+            else:
+                intercept = self.intercept_dist(
+                    name="intercept", **self.model_config["intercept"]["kwargs"]
+                )
+
+            if self.time_varying_media_effect:
+                tv_multiplier_media = time_varying_prior(
+                    name="tv_multiplier_media",
+                    X=time_index,
+                    X_mid=self._time_index_mid,
+                    positive=True,
+                    m=200,
+                    L=[self._time_index_mid + 365 / self._time_resolution],
+                    ls_mu=365 / self._time_resolution * 2,
```

Comment:
Maybe pull this out into a constant
DAYS_IN_YEAR = 365

Maybe this can be consolidated with this hard coded value as well
https://github.com/pymc-labs/pymc-marketing/blob/06b111ebe679ad35822e63a0d96b44195eee6cee/pymc_marketing/mmm/delayed_saturated_mmm.py#L478

Reply:
gotcha